lib/internal/Magento/Framework/Serialize/Serializer/JsonHexTag.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Framework\Serialize\Serializer;
+
+use Magento\Framework\Serialize\SerializerInterface;
+
+/**
+ * Serialize data to JSON, unserialize JSON encoded data
+ *
+ * @api
+ * @since 100.2.0
+ */
+class JsonHexTag extends Json implements SerializerInterface
+{
+    /**
+     * {@inheritDoc}
+     * @since 100.2.0
+     */
+    public function serialize($data): string
+    {
+        $result = json_encode($data, JSON_HEX_TAG);
```

Comment:
There are much more constants to `json_encode`, would it make sense to look for a more flexible solution?

Reply:
I was also thinking that but it would make BC hard. The serializer interface doesn't allow any kind of options. Any suggestions?

Reply:
@quisse Please disregard my previous comment. After further analysis of this I think those options would not be the best approach.

Reply:
@quisse We'll additionally discuss this with core architects, to find an appropriate solution. Please hold on, we are going to get back to it soon.

Reply:
Hi @quisse
It's still in the process of discussion with the architects team. Sorry for taking so long.

Reply:
Thx for the update 😄